lib/temporal/workflow/state_manager.rb

```diff
@@ -241,6 +241,7 @@ def apply_event(event)
           dispatch(target, 'failed', 'StandardError', from_payloads(event.attributes.cause))
 
         when 'CHILD_WORKFLOW_EXECUTION_STARTED'
+          dispatch(target, 'started', event.attributes.workflow_execution)
```

Comment:
There seems to be an issue here — you're dispatching the event with one argument (`event.attributes.workflow_execution`), however the handler is registered without arguments. This suggests that the new behaviour isn't working properly and we need to figure out why the tests are still passing (indicating a potential flake)

Reply:
@arya-k hmm, interesting:

```ruby
>> proc = -> { p 'called' }
=> #<Proc:0x0000000123aa6da0 (irb):1 (lambda)>
>> proc.call('a')
(irb):1:in `block in <main>': wrong number of arguments (given 1, expected 0) (ArgumentError)
```

I wonder if a block is handled differently. But you're right, your example works as expected and just ignores the argument